internal/loader/configuration_client_manager.go

```diff
@@ -492,11 +493,22 @@ func newClientAssertionCredential(ctx context.Context, serviceAccountName string
 		return nil, err
 	}
 
-	clientId := serviceAccountObj.ObjectMeta.Annotations[AnnotationClientID]
-	tenantId := serviceAccountObj.Annotations[AnnotationTenantID]
+	tenantId, ok := os.LookupEnv(strings.ToUpper(AzureTenantId))
```

Comment:
Why the `tenantId` is required?

Reply:
Good question, the `tenantId` is there if enable the cluster workload identity, but need confirm it.

Reply:
Confirmed, once the workload identity is enabled, the `tenantId` should be there, and the existing way we support for workload identity also reply on this `tenantId `

Reply:
If namespaced service account is used, is the value of `tenantId` same as `AnnotationTenantID` from service account? If customers need to explicitly set this value, will users from other namespace can get the `tenantId` through `kubectl describe po k8sproviderpodxxxx`？

Reply:
I don't quite get your question. `tenantId` if for global, each service account could overwrite is by setting `AnnotationTenantID`